src/ring_theory/finiteness.lean

```diff
@@ -301,6 +312,26 @@ begin
     exact ring_hom.ker_coe_equiv (equiv.symm.to_ring_equiv), }
 end
 
+/-- If `A` is a finitely presented `R`-algebra, then `mv_polynomial (fin n) A` is finitely presented
+as `R`-algebra. -/
+lemma mv_polynomial_of_finite_presentation (hfp : finite_presentation R A) (ι : Type*)
+  [fintype ι] : finite_presentation R (_root_.mv_polynomial ι A) :=
+begin
+  obtain ⟨n, e⟩ := fintype.exists_equiv_fin ι,
+  replace e := (mv_polynomial.rename_equiv A (nonempty.some e)).restrict_scalars R,
+  refine equiv _ e.symm,
+  obtain ⟨m, I, e, hfg⟩ := iff.1 hfp,
+  refine equiv _ (mv_polynomial.map_alg_equiv (fin n) e),
+  letI : is_scalar_tower R (_root_.mv_polynomial (fin m) R)
+    (@ideal.map _ (_root_.mv_polynomial (fin n) (_root_.mv_polynomial (fin m) R))
+    _ _ mv_polynomial.C I).quotient := is_scalar_tower.comap,
+  refine equiv _ (alg_equiv.restrict_scalars R (@mv_polynomial.quotient_equiv_quotient_mv_polynomial
+    _ (fin n) _ I)).symm,
```

Comment:
dot notation again:
```suggestion
  refine equiv _ (((@mv_polynomial.quotient_equiv_quotient_mv_polynomial
    _ (fin n) _ I)).restrict_scalars R).symm,
```
although this may have bad parentheses now